### PR TITLE
Added console runner app.  Fix reference to forward version of Newtonsof...

### PR DESCRIPTION
**Pull request by sethwebster**
...t.Json that was

breaking for SignalR

### CODE REVIEW

File: Jabbot.ConsoleBotHost/Program.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        }
+        private static bool TryCreateRoomIfNotExists(string roomName, Bot bot)
+        {
+            try
+            {
+                bot.CreateRoom(roomName);
+            }
+            catch (AggregateException e)
+            {
+                if (!e.InnerExceptions.FirstOrDefault().Message.Contains("exists"))
```
Comment (davidfowl) [inline Jabbot.ConsoleBotHost/Program.cs]: Instead use e.GetBaseException()

**Comment by sethwebster** [comment]
Updated

File: Jabbot.ConsoleBotHost/Program.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Configuration;
+using System.Diagnostics;
```
Comment (davidfowl) [inline Jabbot.ConsoleBotHost/Program.cs]: tsk, usings not sorted :)

File: Jabbot.ConsoleBotHost/Program.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        private static bool TryCreateRoomIfNotExists(string roomName, Bot bot)
+        {
+            try
+            {
+                bot.CreateRoom(roomName);
+            }
+            catch (AggregateException e)
+            {
+                if (!e.GetBaseException().Message.Contains("exists"))
+                    return false;
```
Comment (davidfowl) [inline Jabbot.ConsoleBotHost/Program.cs:53]: Add braces to ifs. Also be explicit about the whole error message, don't just check for contains.

Comment (sethwebster) [inline Jabbot.ConsoleBotHost/Program.cs:53]: Any chance you will change the message? Should we think about a strategy for errors that doesnt couple to user string?

Comment (davidfowl) [inline Jabbot.ConsoleBotHost/Program.cs:53]: Probably need to return error codes + error message to make it more robust.

**Pull request merged**